Pipettes!

BallOfEnergy1 opened the pull request:
Added:
Pipettes

These are able to grab millibucket amounts out of machines, preventing the wasting of fluids and allowing for precision fluid crafting.

(still WIP, will confirm when ready)

BallOfEnergy1 committed:
pipettes!

BallOfEnergy1 committed:
Merge remote-tracking branch 'origin/New_master'

BallOfEnergy1 committed:
Merge branch 'HbmMods:master' into New_master

BallOfEnergy1 committed:
small flixes

BallOfEnergy1 committed:
raaagh

HbmMods commented on src/main/java/com/hbm/items/tool/ItemPipette.java:
```diff
@@ -0,0 +1,175 @@
+
+    @SideOnly(Side.CLIENT) protected IIcon overlayIcon;
+
+    public int amount = this.getMaxFill();
```
there's only one item instance and therefore that value is shared for each pipette, you have to use NBT for that

HbmMods commented on src/main/java/com/hbm/items/tool/ItemPipette.java:
```diff
@@ -0,0 +1,175 @@
+
+    public int amount = this.getMaxFill();
+
+    public FluidType type = Fluids.NONE;
```
same here

HbmMods commented on src/main/java/com/hbm/items/tool/ItemPipette.java:
```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
```
pretty sure that annotation will at best not let this work on servers and at worst crash the game

HbmMods commented on src/main/java/com/hbm/items/tool/ItemPipette.java:
```diff
@@ -0,0 +1,175 @@
+            if (this.getFill(stack, type) == 0) {
+
+                if(this != ModItems.pipette_laboratory)
+                    this.amount = player.isSneaking() ? Math.min(this.amount + 50, 1_000) : Math.max(this.amount - 50, 50);
```
i genuinely can't even tell what this code is supposed to do, or why you're handling the amount at all when you just checked the amount to be 0

BallOfEnergy1 replied:
It's meant to allow the player to change the amount of max fluid in the pipette, probably gonna rename the variable from `amount` to `maxAmount` or something of the like. It checks if the fill is 0, and if so allows the changing of the max capacity in units of either 1mB or 50mB, depending on if it's a laboratory pipette or a normal one.

BallOfEnergy1 committed:
Small fix

BallOfEnergy1 committed:
Merge branch 'HbmMods:master' into New_master

BallOfEnergy1 committed:
2:15 am
forgive my terrible coding

BallOfEnergy1 commented:
NBT fixed,
textures done,
tested on servers,
please review in case I missed anything, its 2 am and I've been working on this for 7 hours.

BallOfEnergy1 committed:
swapping the sneaking control for changing capacity, seems more natural to shift to decrease.

HbmMods commented on src/main/java/com/hbm/crafting/ToolRecipes.java:
```diff
@@ -128,7 +128,10 @@ public static void register() {
 		CraftingManager.addShapelessAuto(new ItemStack(ModItems.sat_relay), new Object[] { ModItems.sat_chip, ModItems.ducttape, ModItems.radar_linker });
-		
+
+		CraftingManager.addRecipeAuto(new ItemStack(ModItems.pipette, 1), new Object[] { "  L", " G ", "G  ", 'L', ModItems.ingot_biorubber, 'G', KEY_CLEARGLASS});
```
`ModItems.ingot_biorubber` should never be used as a recipe ingredient, it always has to be interchangeable with rubber, so use `ANY_RUBBER.ingot()` instead

HbmMods commented on src/main/java/com/hbm/inventory/recipes/AssemblerRecipes.java:
```diff
@@ -696,6 +696,12 @@ private static void registerDefaults() {
 				new ComparableStack(ModItems.motor, 2),
 			}, 200);
 
+		makeRecipe(new ComparableStack(ModItems.pipette_laboratory, 1),new AStack[] {
```
Feels a bit tedious to make in an assembler, especially for what is essentially just a worse pipette. Maybe use the anvil instead? Also, what are the circuits for?

BallOfEnergy1 replied:
The circuits are there because the laboratory pipette is considered a form of "micropipette", which have systems/mechanisms much more complex than normal pipettes.

BallOfEnergy1 committed:
Merge branch 'HbmMods:master' into New_master

BallOfEnergy1 committed:
changed lab pipette from assembler to anvil, fixed rubber shit, should be done
(circuits are because its considered a "micropipette" and is therefore much more complex than a normal pipette.)

BallOfEnergy1 committed:
Merge remote-tracking branch 'origin/New_master'

BallOfEnergy1 commented:
woo!
